docroot/modules/humsci/hs_bugherd/src/Form/HsBugherdForm.php

```diff
@@ -0,0 +1,162 @@
+    $form = parent::buildForm($form, $form_state);
+    $config = $this->config('bugherdapi.settings');
+
+    $form['api_key'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('BugHerd API key'),
+      '#default_value' => $config->get('api_key'),
+      '#required' => TRUE,
+      //'#ajax' => [
+      //  'callback' => '::formAjaxSubmit',
+      //  'wrapper' => 'project-id',
+      //],
+    ];
+
+    $form['project_id'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Bugherd Project'),
+      '#default_value' => $config->get('project_id'),
+      '#options' => $this->bugherdApi->getProjects(),
+      '#prefix' => '<div id="project-id">',
+      '#suffix' => '</div>',
+    ];
+    $form['jira_project'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Jira Project'),
+      '#default_value' => $config->get('jira_project'),
+      '#required' => TRUE,
+    ];
+
+    $form['status_map'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Bugherd to Jira Mapping'),
+      '#tree' => TRUE,
+    ];
+
+    $form['status_map'][HsBugherd::BUGHERDAPI_BACKLOG] = [
+      '#type' => 'number',
+      '#title' => $this->t('Backlog Status'),
+      '#description' => $this->t('The JIRA status code for a "Backlog" status.'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('status_map.' . HsBugherd::BUGHERDAPI_BACKLOG),
+    ];
+
+    $form['status_map'][HsBugherd::BUGHERDAPI_TODO] = [
+      '#type' => 'number',
+      '#title' => $this->t('ToDo Status'),
+      '#description' => $this->t('The JIRA status code for a "To Do" status.'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('status_map.' . HsBugherd::BUGHERDAPI_TODO),
+    ];
+    $form['status_map'][HsBugherd::BUGHERDAPI_DOING] = [
+      '#type' => 'number',
+      '#title' => $this->t('Doing Status'),
+      '#description' => $this->t('The JIRA status code for a "Doing" status.'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('status_map.' . HsBugherd::BUGHERDAPI_DOING),
+    ];
+
+    $form['status_map'][HsBugherd::BUGHERDAPI_DONE] = [
+      '#type' => 'number',
+      '#title' => $this->t('Done Status'),
+      '#description' => $this->t('The JIRA status code for a "Done" status.'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('status_map.' . HsBugherd::BUGHERDAPI_DONE),
+    ];
+
+    $form['status_map'][HsBugherd::BUGHERDAPI_CLOSED] = [
+      '#type' => 'number',
+      '#title' => $this->t('Closed Status'),
+      '#description' => $this->t('The JIRA status code for a "Closed" status. This is normally after the user has accepted the change.'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('status_map.' . HsBugherd::BUGHERDAPI_CLOSED),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
```

Comment:
Some validation that the settings are working would be nice. Like, checking to see if you can connect to bugherd or that the jira project exists and is available to the configured user.

Reply:
Validation added 👍

Reply:
Neato, thank you.